src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -238,15 +238,20 @@ private bool MatchString(string str)
                 pos = runtextpos;
             }
 
+            // Vectorize?
             if (!_caseInsensitive)
             {
-                while (c != 0)
+                if (!str.AsSpan().SequenceEqual(runtext.AsSpan(pos - c, c)))
```

Comment:
We should see some benefit here from the vectorization work that went into `string`

Reply:
there's another one on line 302.

is there a utility to compare spans ignoring case? we have several of those loops too.